lib/spack/spack/package.py

```diff
@@ -1464,7 +1467,27 @@ def do_install(self, **kwargs):
 
     @property
     def test_log_name(self):
-        return 'test-%s-out.txt' % self.spec.format('{name}-{version}-{hash:7}')
+        return 'test-{0}-out.txt' \
+            .format(self.spec.format('{name}-{version}-{hash:7}'))
+
+    @property
+    def test_dir(self):
+        return self.test_stage.join(
+            self.spec.format('{name}-{version}-{hash}'))
+
+    def copy_src_to_install(self, src_subdir, install_subdir):
```

Comment:
Is the point of this method just to wrap an `install` operation with the name of the metadata directory? I'm not really following the importance of this.

Reply:
This is a helper method intended to facilitate copying a source subdirectory to the install metadata directory.  I am assuming it would likely be a frequently used operation as a first step of a package's `run_after` smoke/install test set up process.